extension/content_dashboard.js

```diff
@@ -7,16 +7,73 @@ $(function() {
     airline = AES.getAirlineCode();
     server = AES.getServerName();
     chrome.storage.local.get(['settings'], function(result) {
-        settings = result.settings;
+      settings = result.settings;
 
-        displayDashboard();
+      if ((settings['flightInformationsFixed'] === undefined) || (settings['flightInformationsFixed'] !== 1)) {
+        fixAndDeleteInvalidData();
+      }
+
+      displayDashboard();
+      dashboardHandle();
+      
+      $("#aes-select-dashboard-main").change(function() {
         dashboardHandle();
-        $("#aes-select-dashboard-main").change(function() {
-            dashboardHandle();
-        });
+      });
     });
 });
 
+function fixAndDeleteInvalidData() {
+  chrome.storage.local.get(null, function (entries) {
+    const tasks = [];
+
+    function cleanAndSave(entry, key) {
```

Comment:
Is there a reason this function is nested?

Reply:
it has access to entries then and functionallity belonged to fixAndDeleteInvalidData, nothing special
of course it can be written differently and also moved somewhere else if you want, in content_dashboard.js is already very large anyway
but it's actually only there temporarily, isn't it?

Reply:
We should write code as if it were permanent.

Improving the code base is a long term project :)

Reply:
well, then just rewrite it according to your wishes :)